x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/ExplainLifecycleIT.java

```diff
@@ -207,6 +207,43 @@ public void testExplainIndicesWildcard() throws Exception {
         });
     }
 
+    @SuppressWarnings("unchecked")
+    public void testExplainIndicesDatesAndAges() throws Exception {
+        createNewSingletonPolicy(client(), policy, "delete", DeleteAction.WITH_SNAPSHOT_DELETE, TimeValue.timeValueDays(100));
+        String withOriginationDate = this.index + "-with-origination-date";
+        String withoutOriginationDate = this.index + "-without-origination-date";
+        long originationDate = randomLongBetween(0, System.currentTimeMillis());
+        createIndexWithSettings(
+            client(),
+            withOriginationDate,
+            alias + withOriginationDate,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put(LifecycleSettings.LIFECYCLE_NAME, policy)
+                .put(LifecycleSettings.LIFECYCLE_ORIGINATION_DATE, originationDate)
+        );
+        createIndexWithSettings(
+            client(),
+            withoutOriginationDate,
+            alias + withoutOriginationDate,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put(LifecycleSettings.LIFECYCLE_NAME, policy)
+        );
+
+        assertBusy(() -> {
+            Map<String, Map<String, Object>> explain = explain(client(), this.index + "*", false, false);
+
+            Map<String, Object> explainIndexWithOriginationDate = explain.get(withOriginationDate);
+            assertThat(explainIndexWithOriginationDate.get("age").equals(explainIndexWithOriginationDate.get("index_age")), is(false));
+
+            Map<String, Object> explainIndexWithoutOriginationDate = explain.get(withoutOriginationDate);
+            assertThat(explainIndexWithoutOriginationDate.get("age").equals(explainIndexWithoutOriginationDate.get("index_age")), is(true));
```

Comment:
I don't think this is stable given we use `System.currentTimeMillis()` which would be a "wall clock" read. I suggest we unit test `IndexLifecycleExplainResponse` and maybe we pass in a `timestamp provider` to `getAge` and `getIndexAge` so we can control the values we pass in? 

so we could say `getAge( () -> System.currentTimeMillis() )`.

What do you think?

Reply:
I am not sure I understand what do you mean. Both ages are calculated based on the same "starting" time, the time the index was created, based on that I expect it to be stable.

Reply:
On the other hand I like the idea of a unit test :). Let me think about it.

Reply:
> Both ages are calculated based on the same "starting" time

Reply:
ah yes, I missed that